--- tsync_pps.c	2024-04-24 06:45:21.000000000 +0000
+++ tsync_pps.c	2025-04-24 13:25:18.292908752 +0000
@@ -41,7 +41,11 @@
 		return -ENOMEM;
 	}
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 129)
+	dev_info(&tsync_pps[idx].pps->dev, "tsync PPS source registered\n");
+#else
 	dev_info(tsync_pps[idx].pps->dev, "tsync PPS source registered\n");
+#endif
 
 	return set_hw_int_mask_by_index(tsync_priv, 0, 0);
 }
@@ -50,7 +54,12 @@
 {
 	u32 idx = tsync_priv->slotPosition;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 129)
+	dev_info(&tsync_pps[idx].pps->dev, "tsync pps source unregistered\n");
+#else
 	dev_info(tsync_pps[idx].pps->dev, "tsync pps source unregistered\n");
+#endif
+
 	pps_unregister_source(tsync_pps[idx].pps);
 }
 
