--- tsync.h	2024-10-16 10:44:32.277187113 +0200
+++ tsync.h	2024-10-16 10:44:54.973060429 +0200
@@ -11,6 +11,9 @@
 **
 *******************************************************************************/
 
+// ASTRE : The version used is COLDFIRE
+#define COLDFIRE
+
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -68,7 +71,7 @@
 #define TSYNC_DRV_VERSION_LENGTH        ( 5 )
 
 
-#include "tsync_error_codes.h"
+#include "drast_lkm/tsyncpci/tsync_error_codes.h"
 
     /*
     ** Handle to the TSync board obj available to the user
@@ -1400,7 +1403,7 @@
     } HR_TS_MODE;
 
     /*** Hardware services ***/
-#include "tsync_hw.h"
+#include "drast_lkm/tsyncpci/tsync_hw.h"
 
     /*=========================================================================
     TSYNC MATCH OBJECT
