--- device_init.c	2014-08-28 20:42:35.000000000 +0000
+++ device_init.c	2024-10-11 08:59:01.435681899 +0000
@@ -352,8 +352,13 @@
     BaseAddr = cfg->PCI.rfm2gCsBase;
     if( BaseAddr != 0 )
     {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,0,0)
+        MappedAddr = (RFM2G_ADDR ) ioremap( BaseAddr,
+            cfg->PCI.rfm2gCsWindowSize );
+#else
         MappedAddr = (RFM2G_ADDR ) ioremap_nocache( BaseAddr,
             cfg->PCI.rfm2gCsWindowSize );
+#endif
         if( MappedAddr == 0 )
         {
             /* Error */
@@ -370,8 +375,13 @@
     BaseAddr = cfg->PCI.rfm2gOrBase;
     if( BaseAddr != 0 )
     {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,0,0)
+        MappedAddr = (RFM2G_ADDR) ioremap( BaseAddr,
+            cfg->PCI.rfm2gOrWindowSize );
+#else
         MappedAddr = (RFM2G_ADDR) ioremap_nocache( BaseAddr,
             cfg->PCI.rfm2gOrWindowSize );
+#endif
         if( MappedAddr == 0 )
         {
             /* Error */
