--- proc_page.c	2014-09-09 19:08:44.000000000 +0000
+++ proc_page.c	2024-10-11 08:59:01.439681877 +0000
@@ -199,7 +199,7 @@
 #include <linux/proc_fs.h>
 #include <linux/seq_file.h>
 
-static char myBuff[1048];
+static char myBuff[PAGE_SIZE];
 
 static int rfm2gFSshow(struct seq_file *m, void *v)
 {
@@ -213,6 +213,15 @@
 	return single_open(file, rfm2gFSshow, NULL);
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,0,0)
+struct proc_ops rfm2gFOS =
+{
+	.proc_open = Rfm2gFSOpen,
+	.proc_read = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
+};
+#else
 struct file_operations rfm2gFOS =
 {
 	.owner = THIS_MODULE,
@@ -221,5 +230,6 @@
 	.llseek = seq_lseek,
 	.release = single_release,
 };
+#endif
 
 #endif
