--- rfm2g_kinc.h	2014-08-28 20:33:00.000000000 +0000
+++ rfm2g_kinc.h	2024-10-11 08:59:01.439681877 +0000
@@ -46,8 +46,17 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,18)
 #include <linux/config.h>
 #else
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,10,0)
+/*
+ * generated/autoconf.h is not available on RedHawk 5.4.6 (2.6.18 kernel), 
+ * neither linux/config.h, but after compilation test, it seems it is not
+ * necessay to include any of such thing.
+ * Sebastien Devaux - August 25th 2015
+ */ 
+#else
 #include <generated/autoconf.h>
 #endif
+#endif
 #ifdef CONFIG_SMP
 #define __SMP__
 #endif
