--- apiint.h	2020-01-03 04:22:33.000000000 +0100
+++ apiint.h	2025-01-31 17:59:46.281200200 +0100
@@ -78,6 +78,12 @@
 #include "apinames.h"  /* Only used by trace function */
 #endif
 
+#ifndef _GLOBALS_H_
+#define EXTERN_API extern
+#else
+#define EXTERN_API
+#endif
+
 /**********************************************************************
 *  Swap the two words in a dword, two bytes in a word
 **********************************************************************/
@@ -252,9 +258,9 @@
 		BT_U32BIT addr;
 } BT_USER_ALLOC_COUNT_ENTRY;
 
-BT_USER_ALLOC_COUNT_ENTRY bt_user_alloc_count[MAX_BTA][MAX_BT_USR_ALLOC_ENTRY]; 
+EXTERN_API BT_USER_ALLOC_COUNT_ENTRY bt_user_alloc_count[MAX_BTA][MAX_BT_USR_ALLOC_ENTRY]; 
 	
-BT_U32BIT bt_user_alloc_entry_index[MAX_BTA];
+EXTERN_API BT_U32BIT bt_user_alloc_entry_index[MAX_BTA];
 
 void DumpUserAlloc(
    BT_UINT   cardnum,       // (i) card number (0 based)
@@ -1484,7 +1490,7 @@
    BT_U16BIT regval);            // (i) new value
 
 /****************** HW Register Get functions *********************/
-BT_U32BIT (*vbtGetRegister[MAX_BTA])(
+EXTERN_API BT_U32BIT (*vbtGetRegister[MAX_BTA])(
    BT_UINT   cardnum,             // (i) card number
    BT_U32BIT regnum);             // (i) register number
 
@@ -1497,7 +1503,7 @@
    BT_U32BIT regnum);             // (i) register number
 /*******************************************************************/
 /****************** IQ Header Pointer functions ********************/
-BT_U32BIT (*vbtGetIqHeadPtr[MAX_BTA])(
+EXTERN_API BT_U32BIT (*vbtGetIqHeadPtr[MAX_BTA])(
    BT_UINT cardnum);      // (i) card number (0 based)
 
 BT_U32BIT v6GetIqHeadPtr(
@@ -1507,7 +1513,7 @@
    BT_UINT cardnum);       // (i) card number (0 based)
 /*******************************************************************/
 /****************** BM Header Pointer functions ********************/
-BT_U32BIT (*vbtGetBMHeadPtr[MAX_BTA])(
+EXTERN_API BT_U32BIT (*vbtGetBMHeadPtr[MAX_BTA])(
    BT_UINT cardnum);      // (i) card number (0 based)
 
 BT_U32BIT v6GetBMHeadPtr(
@@ -1517,7 +1523,7 @@
    BT_UINT cardnum);       // (i) card number (0 based)
 /*******************************************************************/
 /************************HW Register Set Functions *****************/
-void (*vbtSetRegister[MAX_BTA])(
+EXTERN_API void (*vbtSetRegister[MAX_BTA])(
    BT_UINT   cardnum,             // (i) card number
    BT_U32BIT regnum,              // (i) register number, offset
    BT_U32BIT regval);             // (i) new value
@@ -1538,7 +1544,7 @@
    BT_U32BIT regval);             // (i) new value
 /******************************************************************/
 /***********************CSC Register Get functions ***************/
-BT_U32BIT (*vbtGetCSCRegister[MAX_BTA])(
+EXTERN_API BT_U32BIT (*vbtGetCSCRegister[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number (0 based)
    BT_U32BIT regnum);        // (i) register number, WORD offset 
 
@@ -1555,7 +1561,7 @@
    BT_U32BIT regnum);       // (i) register number, WORD offset
 /*******************************************************************/
 /***********************CSC Register Set functions******************/
-void (*vbtSetCSCRegister[MAX_BTA])	(
+EXTERN_API void (*vbtSetCSCRegister[MAX_BTA])	(
    BT_UINT   cardnum,       // (i) card number (0 based)
    BT_U32BIT regnum,        // (i) register number, WORD offset
    BT_U32BIT regval);
@@ -1572,7 +1578,7 @@
 /******************************************************************/
 
 /****************** Trigger Register Get functions ****************/
-BT_U32BIT (*vbtGetTrigRegister[MAX_BTA])(
+EXTERN_API BT_U32BIT (*vbtGetTrigRegister[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number (0 based)
    BT_U32BIT regnum);       // (i) register number, WORD offset
 
@@ -1585,7 +1591,7 @@
    BT_U32BIT regnum);       // (i) register number, WORD offset
 /*****************************************************************/
 /******************* Trigger Register Set Functions **************/
-void (*vbtSetTrigRegister[MAX_BTA])(
+EXTERN_API void (*vbtSetTrigRegister[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number (0 based)
    BT_U32BIT regnum,        // (i) register number, WORD offset
    BT_U32BIT regval);       // (i) new value
@@ -1601,7 +1607,7 @@
    BT_U32BIT regval);       // (i) new value
 /******************************************************************/
 /*******************Time Tag Register Get Functions ***************/
-BT_U32BIT (*vbtGetTTRegister[MAX_BTA])(
+EXTERN_API BT_U32BIT (*vbtGetTTRegister[MAX_BTA])(
    BT_UINT   cardnum,             // (i) card number
    BT_U32BIT regnum);             // (i) register number
 
@@ -1614,7 +1620,7 @@
    BT_U32BIT regnum);             // (i) register number
 /*****************************************************************/
 /***************** Time Tag Register Set Functions ***************/
-void (*vbtSetTTRegister[MAX_BTA])(
+EXTERN_API void (*vbtSetTTRegister[MAX_BTA])(
    BT_UINT   cardnum,             // (i) card number
    BT_U32BIT regnum,              // (i) register number, offset
    BT_U32BIT regval);             // (i) new value
@@ -1630,7 +1636,7 @@
    BT_U32BIT regval);             // (i) new value
 /*****************************************************************/
 /*********************** HIF Read Functions **********************************/
-void (*vbtReadGLBR[MAX_BTA])(
+EXTERN_API void (*vbtReadGLBR[MAX_BTA])(
    BT_UINT   cardnum,        // (i) card number
    BT_U32BIT *lpbuffer,      // (o) host buffer (destination)
    BT_U32BIT byteOffset,     // (i) byte offset within adapter memory (source)
@@ -1649,7 +1655,7 @@
    BT_UINT   bytesToRead);   // (i) number of bytes to copy
 /****************************************************************************/
 /********************* Global Register Write Functions **********************************/
-void (*vbtWriteGLBR[MAX_BTA])(
+EXTERN_API void (*vbtWriteGLBR[MAX_BTA])(
    BT_UINT   cardnum,        // (i) card number
    BT_U32BIT *lpbuffer,      // (i) host buffer (source)
    BT_U32BIT byteOffset,     // (o) byte offset within adapter (destination)
@@ -1668,7 +1674,7 @@
    BT_UINT   wordsToWrite);  // (i) number of dword to copy
 /****************************************************************************/
 /******************************* Discrete Registers *************************/
-BT_U32BIT (*vbtGetDiscrete[MAX_BTA])(
+EXTERN_API BT_U32BIT (*vbtGetDiscrete[MAX_BTA])(
    BT_UINT   cardnum,         // (i) card number
    BT_U32BIT regnum);         // (i) register number
 
@@ -1685,7 +1691,7 @@
    BT_U32BIT regnum);         // (i) register number
 /******************************************************************************/
 
-void (*vbtSetDiscrete[MAX_BTA])(
+EXTERN_API void (*vbtSetDiscrete[MAX_BTA])(
    BT_UINT   cardnum,         // (i) card number
    BT_U32BIT regnum,          // (i) register number, offset
    BT_U32BIT regval);         // (i) new value
@@ -1705,7 +1711,7 @@
    BT_U32BIT regnum,          // (i) register number, offset
    BT_U32BIT regval);         // (i) new value
 /******************************************************************************/
-void (*vbtReadTimeTag[MAX_BTA])(
+EXTERN_API void (*vbtReadTimeTag[MAX_BTA])(
    BT_UINT   cardnum,         // (i) card number
    BT_U32BIT * timetag);      // (o) resulting 48-bit time value from HW
 
@@ -1722,7 +1728,7 @@
    BT_U32BIT * timetag);      // (o) resulting 48-bit time value from HW
 
 /********************** Time Tag Write functions ***************************/
-void (*vbtWriteTimeTag[MAX_BTA])(
+EXTERN_API void (*vbtWriteTimeTag[MAX_BTA])(
    BT_UINT   cardnum,         // (i) card number
    BT1553_TIME * timetag);    // (i) 48-bit time value to load into register
 
@@ -1739,7 +1745,7 @@
    BT1553_TIME * timetag);    // (i) 48-bit time value to load into register
 /********************* Time Tag Read functions ******************************/
 
-void (*vbtWriteTimeTagIncr[MAX_BTA])(
+EXTERN_API void (*vbtWriteTimeTagIncr[MAX_BTA])(
    BT_UINT   cardnum,        // (i) card number
    BT_U32BIT incr);          // (i) 32-bit time increment
 
@@ -1755,7 +1761,7 @@
    BT_UINT   cardnum,        // (i) card number
    BT_U32BIT incr);          // (i) 32-bit time increment
 /****************************************************************************/
-void (*vbtReadRAM[MAX_BTA])(
+EXTERN_API void (*vbtReadRAM[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number
    BT_U16BIT *lpbuffer,     // (o) host buffer (destination)
    BT_U32BIT byteOffset,    // (i) byte offset within adapter memory (source)
@@ -1780,7 +1786,7 @@
    BT_UINT   bytesToRead);  // (i) number of bytes to copy
 /*******************************************************************************/
 /****************************************************************************/
-void (*vbtReadBMRAM[MAX_BTA])(
+EXTERN_API void (*vbtReadBMRAM[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number
    BT_U16BIT *lpbuffer,     // (o) host buffer (destination)
    BT_U32BIT byteOffset,    // (i) byte offset within adapter memory (source)
@@ -1793,7 +1799,7 @@
    BT_UINT   bytesToRead);  // (i) number of bytes to copy
 /*******************************************************************************/
 
-void (*vbtReadRelRAM[MAX_BTA])(
+EXTERN_API void (*vbtReadRelRAM[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number
    BT_U16BIT *lpbuffer,     // (o) host buffer (destination)
    BT_U32BIT byteOffset,    // (i) byte offset within adapter memory (source)
@@ -1812,7 +1818,7 @@
    BT_UINT   bytesToRead);  // (i) number of bytes to copy
 
 /************************ 32-BIT RAM Read *************************************/
-void (*vbtReadRAM32[MAX_BTA])(
+EXTERN_API void (*vbtReadRAM32[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number
    BT_U32BIT *lpbuffer,     // (o) host buffer (dest)
    BT_U32BIT memoffset,     // (i) memory offset within adapter memory (source)
@@ -1843,7 +1849,7 @@
    BT_UINT   numbytes);     // (i) number of bytes to copy
 /*******************************************************************************/
 /************************ 32-BIT BM RAM Read *************************************/
-void (*vbtReadBMRAM32[MAX_BTA])(
+EXTERN_API void (*vbtReadBMRAM32[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number
    BT_U32BIT *lpbuffer,     // (o) host buffer (dest)
    BT_U32BIT memoffset,     // (i) memory offset within adapter memory (source)
@@ -1857,7 +1863,7 @@
 
 /*******************************************************************************/
 /************************ 32-BIT BM RAM Read *************************************/
-void (*vbtWriteBMRAM32[MAX_BTA])(
+EXTERN_API void (*vbtWriteBMRAM32[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number
    BT_U32BIT *lpbuffer,     // (o) host buffer (dest)
    BT_U32BIT memoffset,     // (i) memory offset within adapter memory (source)
@@ -1872,7 +1878,7 @@
 /*******************************************************************************/
 
 /************************ Read Interrupt Queue *********************************/
-void (*vbtReadIntQueue[MAX_BTA])(
+EXTERN_API void (*vbtReadIntQueue[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number
    BT_U32BIT *lpbuffer,     // (o) host buffer (dest)
    BT_U32BIT memoffset);    // (i) memory offset within adapter memory (source)
@@ -1888,7 +1894,7 @@
    BT_U32BIT memoffset);     // (i) memory offset within adapter memory (source)
 /*******************************************************************************/
 
-void (*vbtReadRelRAM32[MAX_BTA])(
+EXTERN_API void (*vbtReadRelRAM32[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number
    BT_U32BIT *lpbuffer,     // (o) host buffer (dest)
    BT_U32BIT memoffset,     // (i) memory offset within adapter memory (source)
@@ -1906,7 +1912,7 @@
    BT_U32BIT memoffset,     // (i) memory offset within adapter memory (source)
    BT_UINT   numbytes);     // (i) number of bytes to copy
 /********************************************************************************/
-void (*vbtWriteRAM[MAX_BTA])(
+EXTERN_API void (*vbtWriteRAM[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number
    BT_U16BIT *lpbuffer,     // (i) host buffer (source)
    BT_U32BIT byteOffset,    // (o) byte offset within adapter (destination)
@@ -1930,7 +1936,7 @@
    BT_U32BIT byteOffset,    // (o) byte offset within adapter (destination)
    BT_UINT   bytesToWrite); // (i) number of bytes to copy
 /*******************************************************************************/
-void (*vbtWriteRelRAM[MAX_BTA])(
+EXTERN_API void (*vbtWriteRelRAM[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number
    BT_U16BIT *lpbuffer,     // (i) host buffer (source)
    BT_U32BIT byteOffset,    // (o) byte offset within adapter (destination)
@@ -1948,7 +1954,7 @@
    BT_U32BIT byteOffset,    // (o) byte offset within adapter (destination)
    BT_UINT   bytesToWrite); // (i) number of bytes to copy
 /******************************************************************************/
-void (*vbtWriteRAM32[MAX_BTA])(
+EXTERN_API void (*vbtWriteRAM32[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number
    BT_U32BIT *lpbuffer,     // (i) host buffer (source)
    BT_U32BIT memoffset,     // (o) memory offset within adapter memory (dest)
@@ -1978,7 +1984,7 @@
    BT_U32BIT memoffset,     // (o) memory offset within adapter memory (dest)
    BT_UINT   numWords);     // (i) number of bytes to copy
 /********************************************************************************/
-void (*vbtWriteRelRAM32[MAX_BTA])(
+EXTERN_API void (*vbtWriteRelRAM32[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number
    BT_U32BIT *lpbuffer,     // (i) host buffer (source)
    BT_U32BIT memoffset,     // (o) memory offset within adapter memory (dest)
@@ -1997,7 +2003,7 @@
    BT_UINT   numWords);     // (i) number of bytes to copy
 /***********************************************************************************/
 /********************************************************************************/
-void (*vbtWriteSharedMemory[MAX_BTA])(
+EXTERN_API void (*vbtWriteSharedMemory[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number
    BT_U32BIT *lpbuffer,     // (i) host buffer (source)
    BT_U32BIT memoffset,     // (o) memory offset within adapter memory (dest)
@@ -2017,7 +2023,7 @@
 /***********************************************************************************/
 /*******************************************************************************/
 
-void (*vbtReadSharedMemory[MAX_BTA])(
+EXTERN_API void (*vbtReadSharedMemory[MAX_BTA])(
    BT_UINT   cardnum,       // (i) card number
    BT_U32BIT *lpbuffer,     // (o) host buffer (destination)
    BT_U32BIT byteOffset,    // (i) byte offset within adapter memory (source)
@@ -2131,7 +2137,7 @@
 
 void vbtShutdown(BT_UINT cardnum);   // (i) card number
 
-void (*vbtNotify[MAX_BTA]) (
+EXTERN_API void (*vbtNotify[MAX_BTA]) (
    BT_UINT cardnum,                   // (i) card number of board/channel to process
    BT_U32BIT   *bmLocalRecordTime);   // (i) Time to call BM recorder if > BM_RECORD_TIME
 void v5Notify(
@@ -2251,7 +2257,7 @@
 void SetFunctionsPTR(BT_UINT cardnum);
 
 //void (*vbtSetCSCRegister[MAX_BTA])	(
-NOMANGLE BT_INT (CCONV *BT_BC_Init[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BC_Init[MAX_BTA]) (
    BT_UINT   cardnum,       // (i) card number (0 based)
    BT_UINT   bc_options,    // (i) Gap Time from start = 0xf; Msg Schd = 0x10; 
                             //     Frame Timing 0x20, no prediction logic 0x40
@@ -2284,7 +2290,7 @@
    BT_U32BIT frame,         // (i) minor frame period, in microseconds
    BT_UINT   num_buffers);   // (i) number of BC message buffers ( 1 or 2 for legacy)
 
-NOMANGLE BT_INT (CCONV *BT_BC_MessageAlloc[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BC_MessageAlloc[MAX_BTA]) (
    BT_UINT cardnum,         // (i) card number (0 - based)
    BT_UINT count);          // (i) number of BC messages to allocate
 
@@ -2296,7 +2302,7 @@
    BT_UINT cardnum,         // (i) card number (0 - based)
    BT_UINT count);          // (i) number of BC messages to allocate
 
-NOMANGLE BT_INT (CCONV *BT_BC_MessageNoop[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BC_MessageNoop[MAX_BTA]) (
    BT_UINT   cardnum,       // (i) card number (0 based)
    BT_UINT messageid,       // (i) Number of BC message to modify
    BT_UINT NoopFlag,        // (i) Flag, 1 -> Set to NOOP, 0 -> Return to msg
@@ -2317,7 +2323,7 @@
                             //           0xf -> Timed Noop
    BT_UINT WaitFlag);       // (i) Flag, 1 -> Wait for BC to not be executing msg,
                             //           0 -> Do not test for BC executing msg.
-NOMANGLE BT_INT (CCONV *BT_BC_MessageRead[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BC_MessageRead[MAX_BTA]) (
    BT_UINT   cardnum,          // (i) card number (0 based)
    BT_UINT mblock_id,          // (i) Number of BC message
    API_BC_MBUF * api_message); // (i) Pointer to buffer to receive msg
@@ -2332,7 +2338,7 @@
    BT_UINT mblock_id,          // (i) Number of BC message
    API_BC_MBUF * api_message); // (i) Pointer to buffer to receive msg
 
-NOMANGLE BT_INT (CCONV *BT_BC_MessageWrite[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BC_MessageWrite[MAX_BTA]) (
    BT_UINT   cardnum,           // (i) card number (0 based)
    BT_UINT   messageid,         // (i) BC Message number
    API_BC_MBUF * api_message);  // (i) pointer to user's buffer containing msg
@@ -2347,7 +2353,7 @@
    BT_UINT   messageid,         // (i) BC Message number
    API_BC_MBUF * api_message);  // (i) pointer to user's buffer containing msg
 
-NOMANGLE BT_INT (CCONV *BT_BC_MessageUpdate[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BC_MessageUpdate[MAX_BTA]) (
    BT_UINT   cardnum,       // (i) card number (0 based)
    BT_UINT   mblock_id,     // (i) BC Message number
    BT_U16BIT * buffer);     // (i) pointer to user's data buffer
@@ -2362,7 +2368,7 @@
    BT_UINT   mblock_id,     // (i) BC Message number
    BT_U16BIT * buffer);     // (i) pointer to user's data buffer
 
-NOMANGLE BT_INT (CCONV *BT_BC_MessageUpdateBuffer[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BC_MessageUpdateBuffer[MAX_BTA]) (
    BT_UINT   cardnum,       // (i) card number (0 based)
    BT_UINT   mblock_id,     // (i) BC Message number
    BT_UINT   buffer_id,     // (i) BC data buffer 0=A 1=B
@@ -2380,7 +2386,7 @@
    BT_UINT   buffer_id,     // (i) BC data buffer 0=A 1=B
    BT_U16BIT * buffer);     // (i) pointer to user's data buffer
 
-NOMANGLE BT_INT (CCONV *BT_BC_Start[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BC_Start[MAX_BTA]) (
    BT_UINT   cardnum,       // (i) card number (0 based)
    BT_UINT messageid);      // (i) BC Message number to start processing at
 
@@ -2392,7 +2398,7 @@
    BT_UINT   cardnum,       // (i) card number (0 based)
    BT_UINT messageid);      // (i) BC Message number to start processing at
 
-NOMANGLE BT_INT (CCONV *BT_BC_ReadNextMessage[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BC_ReadNextMessage[MAX_BTA]) (
    BT_INT cardnum,
    BT_UINT timeout,
    BT_INT rt_addr,
@@ -2416,7 +2422,7 @@
    BT_INT tr, 
    API_BC_MBUF *pBC_mbuf);
 
-NOMANGLE BT_INT (CCONV *BT_BC_ReadlastMessage[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BC_ReadlastMessage[MAX_BTA]) (
    BT_INT cardnum,
    BT_INT rt_addr,
    BT_INT subaddress,
@@ -2437,7 +2443,7 @@
    BT_INT tr,
    API_BC_MBUF *pBC_mbuf);
 
-NOMANGLE BT_INT (CCONV *BT_BC_ReadLastMessageBlock[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BC_ReadLastMessageBlock[MAX_BTA]) (
    BT_INT cardnum,
    BT_INT rt_addr_mask,
    BT_INT subaddr_mask,
@@ -2464,7 +2470,7 @@
 /**************************************************************
  *  RT functions
  *************************************************************/
-NOMANGLE BT_INT (CCONV *BT_RT_Init[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_RT_Init[MAX_BTA]) (
     BT_UINT cardnum,          // (i) card number
     BT_UINT testflag);        // (i) flag; ignored!
 
@@ -2476,7 +2482,7 @@
     BT_UINT cardnum,          // (i) card number
     BT_UINT testflag);        // (i) flag; ignored!
 
-NOMANGLE BT_INT (CCONV *BT_RT_AbufRead[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_RT_AbufRead[MAX_BTA]) (
    BT_UINT cardnum,           // (i) card number
    BT_UINT rtaddress,         // (i) RT Terminal Address
    API_RT_ABUF * abuf);       // (o) Pointer to buffer to receive abuf data
@@ -2491,7 +2497,7 @@
    BT_UINT rtaddress,         // (i) RT Terminal Address
    API_RT_ABUF * abuf);       // (o) Pointer to buffer to receive abuf data
 
-NOMANGLE BT_INT (CCONV *BT_RT_AbufWrite[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_RT_AbufWrite[MAX_BTA]) (
    BT_UINT cardnum,           // (i) card number
    BT_UINT rtaddress,         // (i) RT Terminal Address
    API_RT_ABUF * abuf);       // (i) Pointer to user's abuf structure
@@ -2505,8 +2511,23 @@
    BT_UINT cardnum,           // (i) card number
    BT_UINT rtaddress,         // (i) RT Terminal Address
    API_RT_ABUF * abuf);       // (i) Pointer to user's abuf structure
+   
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_RT_QuickEnable[MAX_BTA]) (
+   BT_UINT cardnum,           // (i) card number
+   BT_UINT rtaddress,         // (i) RT Terminal Address
+   BT_U16BIT enableA, BT_U16BIT enableB);       // (i) Pointer to user's abuf structure
+
+NOMANGLE BT_INT CCONV v5_RT_QuickEnable(
+   BT_UINT cardnum,           // (i) card number
+   BT_UINT rtaddress,         // (i) RT Terminal Address
+   BT_U16BIT enableA, BT_U16BIT enableB);       // (i) Pointer to user's abuf structure
+
+NOMANGLE BT_INT CCONV v6_RT_QuickEnable(
+   BT_UINT cardnum,           // (i) card number
+   BT_UINT rtaddress,         // (i) RT Terminal Address
+   BT_U16BIT enableA, BT_U16BIT enableB);       // (i) Pointer to user's abuf structure
 
-NOMANGLE BT_INT (CCONV *BT_RT_CbufbroadWrite[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_RT_CbufbroadWrite[MAX_BTA]) (
    BT_UINT cardnum,           // (i) card number
    BT_UINT subaddr,           // (i) Subaddress to write
    BT_UINT tr,                // (i) Non-zero indicates Transmit, zero Receive
@@ -2524,7 +2545,7 @@
    BT_UINT tr,                // (i) Non-zero indicates Transmit, zero Receive
    API_RT_CBUFBROAD * apicbuf); // (i) Buffer containing Control Buffer
 
-NOMANGLE BT_INT (CCONV *BT_RT_CbufWrite[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_RT_CbufWrite[MAX_BTA]) (
    BT_UINT cardnum,           // (i) card number
    BT_UINT rtaddr,            // (i) Remote Terminal address to be setup
    BT_UINT subaddr,           // (i) Subaddress to be setup
@@ -2551,7 +2572,7 @@
                               //     Current buffer number if second call.
    API_RT_CBUF * apicbuf);    // (i) User-supplied CBUF; legal word count mask
 
-NOMANGLE BT_INT (CCONV *BT_RT_MessageBufferNext[MAX_BTA])(
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_RT_MessageBufferNext[MAX_BTA])(
    BT_UINT cardnum,            // (i) card number
    BT_UINT rtaddr,             // (i) RT Terminal Address of message to read
    BT_UINT subaddr,            // (i) RT Subaddress of message to read
@@ -2559,7 +2580,7 @@
    BT_UINT *mbuf_id  );        // (i) ID number of message buffer (0-based)
   
 
-NOMANGLE BT_INT (CCONV *BT_RT_MessageRead[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_RT_MessageRead[MAX_BTA]) (
    BT_UINT cardnum,            // (i) card number
    BT_UINT rtaddr,             // (i) RT Terminal Address of message to read
    BT_UINT subaddr,            // (i) RT Subaddress of message to read
@@ -2597,7 +2618,7 @@
    BT_UINT mbuf_id,            // (i) ID number of message buffer (0-based)
    API_RT_MBUF_READ * apimbuf);// (o) User's buffer to receive data
 
-NOMANGLE BT_INT (CCONV *BT_RT_MessageWriteDef[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_RT_MessageWriteDef[MAX_BTA]) (
    BT_UINT cardnum,              // (i) card number
    BT_UINT rtaddr,               // (i) RT Terminal Address
    API_RT_MBUF_WRITE * apimbuf); // (i) Pointer to user's RT Message Buffer
@@ -2614,7 +2635,7 @@
 
 // Bus Monitor functions
 
-NOMANGLE BT_INT (CCONV *BT_BM_FilterWrite[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BM_FilterWrite[MAX_BTA]) (
    BT_UINT   cardnum,       // (i) card number (0 based)
    BT_UINT   rtaddr,
    BT_UINT   subaddr,
@@ -2635,7 +2656,7 @@
    BT_UINT   tr,
    API_BM_CBUF * api_cbuf);
 
-NOMANGLE BT_INT (CCONV *BT_BM_MessageAlloc[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BM_MessageAlloc[MAX_BTA]) (
    BT_UINT   cardnum,       // (i) card number (0 based)
    BT_UINT   mbuf_count,    // Message count converted to bytes;
    BT_UINT   * mbuf_actual, // Actual allocated count 
@@ -2653,7 +2674,7 @@
    BT_UINT   * mbuf_actual, // Actual allocated count 
    BT_U32BIT enable);       // interrupt enables
 
-NOMANGLE BT_INT (CCONV *BT_BM_StartStop[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BM_StartStop[MAX_BTA]) (
    BT_UINT   cardnum,       // (i) card number (0 based)
    BT_UINT startflag);      // (i) 0=stop
                             //     1=start
@@ -2674,7 +2695,7 @@
                             //     3=start + reset time tag
                             //     0xf BM_START_BIT
 
-NOMANGLE BT_INT (CCONV *BT_BM_ReadNextMessage[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BM_ReadNextMessage[MAX_BTA]) (
    int cardnum,BT_UINT timeout,
    BT_INT rt_addr,  
    BT_INT subaddress,
@@ -2695,7 +2716,7 @@
    BT_INT tr, 
    API_BM_MBUF *pBM_mbuf);
 
-NOMANGLE BT_INT (CCONV *BT_BM_ReadLastMessage[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BM_ReadLastMessage[MAX_BTA]) (
    int cardnum,
    BT_INT rt_addr,
    BT_INT subaddress,
@@ -2716,7 +2737,7 @@
    BT_INT tr, 
    API_BM_MBUF *pBM_mbuf);
 
-NOMANGLE BT_INT (CCONV *BT_BM_ReadLastMessageBlock[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BM_ReadLastMessageBlock[MAX_BTA]) (
    int cardnum,
    BT_INT rt_addr_mask,
    BT_INT subaddr_mask,
@@ -2740,7 +2761,7 @@
    BT_UINT *mcount,
    API_BM_MBUF *pBM_mbuf);
 
-NOMANGLE BT_INT (CCONV *BT_BM_Init[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_BM_Init[MAX_BTA]) (
    BT_UINT   cardnum,       // (i) card number (0 based)
    BT_UINT enable_a,
    BT_UINT enable_b);
@@ -2756,7 +2777,7 @@
    BT_UINT enable_b);
 
 // Misc functions
-NOMANGLE BT_INT (CCONV *BT_DiscreteTriggerOut[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_DiscreteTriggerOut[MAX_BTA]) (
    BT_UINT cardnum,         // (i) card number
    BT_INT disflag);         // (i) select one or more output trigger
 
@@ -2770,7 +2791,7 @@
    BT_UINT cardnum,         // (i) card number
    BT_INT disflag);         // (i) select one or more output trigger
 
-NOMANGLE BT_INT (CCONV *BT_DiscreteTriggerIn[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_DiscreteTriggerIn[MAX_BTA]) (
    BT_UINT cardnum,         // (i) card number
    BT_INT trigflag);        // (i) 0=none, 1=485, 2=dis7 3=dis8 (v5)
                             // (i) 0=none,  1 to  31 discrete channel (v6)
@@ -2785,7 +2806,7 @@
    BT_INT trigflag);        // (i) 0=none,  1 to  31 discrete channel
                             //             -1 to -31 differential channel
 
-NOMANGLE BT_INT (CCONV *BT_SetOptions[MAX_BTA]) (
+EXTERN_API NOMANGLE BT_INT (CCONV *BT_SetOptions[MAX_BTA]) (
    BT_UINT cardnum,        // (i) card number
    BT_UINT intflag,        // (i) 0x0001 -> suppress MF OFlow message
                            //     0x0002 -> Monitor Invalid Commands
