--- btdrv.h	2020-01-03 03:22:33.000000000 +0000
+++ btdrv.h	2024-10-16 12:58:02.796284950 +0000
@@ -78,7 +78,7 @@
 /******************************************************************************/
 
 /* Define the shift counts used to convert a board offset into a board frame: */
-BT_UINT BT_PCI_MEMORY[MAX_BTA];                    /* Number of 1KB blocks of memory   */
+EXTERN BT_UINT BT_PCI_MEMORY[MAX_BTA];                    /* Number of 1KB blocks of memory   */
 #define BT_VME_MEMORY          1024       /* Number of 1KB blocks of memory   */
 
 /* Define the base address of the RT segment */
@@ -90,8 +90,8 @@
 #define HIR_AC_REG         1     /* Additional capabilties register for QPM   */
 
 // Definitions for the default board...byte offsets from the base address:
-BT_UINT RAM_SIZE[MAX_BTA];               /* Default RAM size                  */
-BT_UINT RAM_OFFSET[MAX_BTA];             /* Offset to 1553 data RAM           */
+EXTERN BT_UINT RAM_SIZE[MAX_BTA];               /* Default RAM size                  */
+EXTERN BT_UINT RAM_OFFSET[MAX_BTA];             /* Offset to 1553 data RAM           */
 
 #define REG_OFFSET          0x30000      /* Offset to start of Register       */ 
 #define HWREG_OFFSET        0x0          /* Offset to 1553 H/W Register       */
