--- busapi.h	2020-01-03 04:22:33.000000000 +0100
+++ busapi.h	2025-01-27 15:57:33.631207773 +0100
@@ -2651,6 +2651,12 @@
 *  RT Function Prototypes
 **********************************************************************/
 
+// ASTRE addon to permit the enable/disable of RT very quick.
+NOMANGLE BT_INT CCONV BusTools_RT_QuickEnable(
+   BT_UINT cardnum,         // (i) card number (0 - based)
+   BT_UINT rtaddress,       // (i) RT address (0 - based)
+   BT_U16BIT enableA, BT_U16BIT enableB);
+
 NOMANGLE BT_INT CCONV BusTools_RT_AbufRead(
    BT_UINT cardnum,         // (i) card number (0 - based)
    BT_UINT rtaddress,       // (i) RT address (0 - based)
