--- notify.c	2020-01-03 03:22:33.000000000 +0000
+++ notify.c	2024-10-16 13:11:21.083829069 +0000
@@ -537,12 +537,17 @@
          }
       }
 
-      status = CEI_WAIT_FOR_EVENT(&sIntFIFO->hEvent, sIntFIFO->dwMilliseconds, &sIntFIFO->mutex);
-      if(status < 0) // Wait error condition skip data processing
-      {
-         channel_status[cardnum].err_info=0x4;
-         continue;
+      // change in Mutex lock/unlock to fix issue ASTRE-4713
+      CEI_MUTEX_LOCK(&sIntFIFO->mutex);
+      if (sIntFIFO->tail_index == sIntFIFO->head_index) {
+         status = CEI_WAIT_FOR_EVENT_NO_LOCK(&sIntFIFO->hEvent, sIntFIFO->dwMilliseconds, &sIntFIFO->mutex);
+         if(status < 0) // Wait error codition skip data processing
+         {
+            channel_status[cardnum].err_info=0x4;
+            continue;
+         }
       }
+      CEI_MUTEX_UNLOCK(&sIntFIFO->mutex);
       channel_status[cardnum].err_info=0x0;
       sIntFIFO->timeout = status;
 
@@ -685,9 +690,12 @@
          sIntFIFO->fifo[head_index].subaddress = subaddress;
          sIntFIFO->fifo[head_index].wordcount  = wordCount;
          sIntFIFO->fifo[head_index].bufferID   = nBufferNum; // Buffer or message number
+         // change in Mutex lock/unlock to fix issue ASTRE-4713
+         CEI_MUTEX_LOCK(&sIntFIFO->mutex);
          sIntFIFO->head_index = head;                        // Update head pointer
          // FIFO has been updated, signal the thread.
          CEI_EVENT_SIGNAL(&sIntFIFO->hEvent,&sIntFIFO->mutex);
+         CEI_MUTEX_UNLOCK(&sIntFIFO->mutex);
       }
    }
 }
